File: TsinghuaNet.UWP.Background/LiveTileTask.vb
Imports TsinghuaNet.Helper
Imports TsinghuaNet.UWP.Helper
Imports Windows.ApplicationModel.Background

Public NotInheritable Class LiveTileTask
    Implements IBackgroundTask

    Public Async Sub Run(taskInstance As IBackgroundTaskInstance) Implements IBackgroundTask.Run
        Dim deferral = taskInstance.GetDeferral()
        Try
            Dim status As New InternetStatus
            Await status.RefreshAsync()
            Dim helper = ConnectHelper.GetHelper(Await status.SuggestAsync())
            If helper IsNot Nothing Then
                Using helper
                    Dim user As FluxUser = Await helper.GetFluxAsync()
                    NotificationHelper.UpdateTile(user)
                    If SettingsHelper.FluxLimit IsNot Nothing Then
                        NotificationHelper.SendWarningToast(user, SettingsHelper.FluxLimit)
                    End If
                End Using
            End If
        Finally
            deferral.Complete()
        End Try
    End Sub
End Class

File: TsinghuaNet.Uno/TsinghuaNet.Uno.UWP.Background/LiveTileTask.cs
using TsinghuaNet.Models;
using TsinghuaNet.Uno.Helpers;
using Windows.ApplicationModel.Background;

namespace TsinghuaNet.Uno.UWP.Background
{
    public sealed class LiveTileTask : IBackgroundTask
    {
        public async void Run(IBackgroundTaskInstance taskInstance)
        {
            var deferral = taskInstance.GetDeferral();
            try
            {
                InternetStatus status = new InternetStatus();
                NetUnoSettings settings = new NetUnoSettings();
                settings.LoadSettings();
                await status.RefreshAsync();
                var helper = ConnectHelper.GetHelper(await status.SuggestAsync());
                if (helper != null)
                {
                    FluxUser user = await helper.GetFluxAsync();
                    NotificationHelper.UpdateTile(user);
                    if (settings.EnableFluxLimit)
                        NotificationHelper.SendWarningToast(user, settings.FluxLimit);
                }
            }
            finally
            {
                deferral.Complete();
            }
        }
    }
}
